### ISSUE TITLE
Process system requirements

### ISSUE THREAD

**Issue by AlanCoding** [ansible/ansible-builder#50]
for my own notes, from this comment:

https://github.com/ansible/ansible-builder/issues/22#issuecomment-650206193

https://opendev.org/opendev/system-config/src/branch/master/docker/python-builder/scripts/assemble

the critical syntax looks like:

```
bindep -l newline >> /output/bindep/run.txt || true
yum -y install $(cat /output/bindep/run.txt)
```

(we may or may not use the intermediary file)

In my testing, the `-q` option (for quiet) is also needed, and very important, a newline is needed at the end of the `bindep.txt` file.

This will be implemented as a `RUN` in the image.

**Comment by AlanCoding** [comment]
This is implemented for a user's file, but not yet for collecting bindep files from collections themselves.

**Comment by AlanCoding** [comment]
Also need to update the `README.md`

**Comment by AlanCoding** [comment]
Updating and documenting this issue...

We merged #52 which addressed this for user-specified bindep file, meaning the `execution-environment.yml` file had an entry `system: bindep.txt` in it. The WIP #59 addresses it for collections. Here is where I'm at with that, the flow loosely reflects the order in which my own thought process shaped up.

The aggressive proposal is most similar to how the similar problem for python requirements was addressed in #57. However, I would, right away, push _against_ this option because bindep stuff has fairly different issues from python stuff. We for bindep, we only really need to detect if the _final result_ is blank or not blank. Including entries which are obviously not relevant isn't a problem, and it's kind of part of the design. Not mentioned here - we may still like to scrub a combined `bindep.txt` file for _literal duplicates_, because this case will occur commonly in the existing collections, like `python3-devel [test platform:rpm]`, which is copy+pasted for several collections like `frr.frr`, `arista.eos`, and many others.

My preference is clearly the in-between proposal.

#### Current PR:

 - in introspect script, exclude lines which are blank or have "test" in them, only "count" bindep.txt if it's not blank after that
 - introspect script returns `bindep.txt` file locations
 - Do a single-line RUN layer in Containerfile for each `bindep.txt` file from user or collections
 - (this can still error if bindep reqs are not relevant to platform)

#### Agressive proposal:

 - in introspect script, only remove obviously blank lines or comment lines
 - introspect script returns lines from `bindep.txt` files
 - Add bindep as a requirement of `ansible-builder` itself
 - import functions from bindep in order to scrub the requirements, remove those which are obviously N/A, test profile in particular
 - builder writes combined `bindep.txt` to build context
 - Do a single-line RUN layer in Containerfile for single `bindep.txt`
 - (not airtight, can still error if bindep reqs are not relevant to platform)

#### In-between proposal:

 - in introspect script, only remove obviously blank lines or comment lines
 - introspect script returns lines from `bindep.txt` files
 - builder writes combined `bindep.txt` to build context (will contain test profile lines, don't care)
 - use multiple RUN layers to install that single `bindep.txt` file:
   - pipe bindep result to intermediate requirements file
   - have some conditional inside Containerfile itself to only run dnf if intermediate requirements file is not blank

#### Verbose proposal:

 - same as in-between proposal, but don't bother combining `bindep.txt`, have multiple lines for each collection `bindep.txt `file

**Comment by AlanCoding** [comment]
@kdelee test coverage in

https://github.com/ansible/ansible-builder/blob/6d6ba5d5934d84d546300f31cfe6048a78555574/test/integration/test_build.py#L72

and 

https://github.com/ansible/ansible-builder/blob/6d6ba5d5934d84d546300f31cfe6048a78555574/test/integration/test_build.py#L60
